lib/fpm/package/rpm.rb

```diff
@@ -562,8 +562,8 @@ def epoch
 
   def to_s(format=nil)
     if format.nil?
-      return super("NAME-VERSION-ITERATION.DIST.ARCH.TYPE").gsub('DIST', attributes[:rpm_dist]) if attributes[:rpm_dist]
-      return super("NAME-VERSION-ITERATION.ARCH.TYPE")
+      return super("NAME-VERSION-ITERATION.DIST.ARCH.EXTENSION").gsub('DIST', attributes[:rpm_dist]) if attributes[:rpm_dist]
```

Comment:
Anyway, if ``format`` argument is provided and is not ``nil``  and f already contains ``"DIST"``, the ``"DIST"`` will not be substituted.